### ISSUE TITLE
Feature: Always On Top visibility option

### ISSUE THREAD

**Issue by marcosilva85** [dangvd/crystal-dock#33]
Is it possible to make crystal-dock stay on top of application windows or even KDE plasma panels ?
I'm not seeking for the "auto-hide" option, instead I'd like to have the dock always present and without creating a margin that offsets maximized applications.

Example of usages:
a) Crystal-dock living on the right side of screen (since it typically does not impact apps).
b) KDE default plasma panel with the icons-task-manager widget removed and crystal-dock centered on top of that panel to replicate the widget's function.

using wayland I could not find this basic feature but it could be that I simply could not discover where it is in settings...

**Comment by dangvd** [comment]
Hi,

This feature was actually available in Crystal Dock v1.x for X11, but for v2.x for Wayland, I had to remove it due to some technical issues.

Specifically, on X11, the dock resizes to small size when minimized and resizes to big size when zoomed in. On Wayland, resizing would cause graphical glitch so now I have to keep the size big all the time and when minimized only make a small part of it visible. So if I enable Always On Top now, the dock will take too much space and get into your way -- for example, it will capture your mouse/keyboard input in the big area even though it's invisible in the outer part. I tried using widget masking to enable mouse/keyboard input only in the small area but it didn't work for some reason.

Anyway I will keep this in mind and try again in the future.
